property/index.js

```diff
@@ -75,6 +77,9 @@ module.exports = yeoman.Base.extend({
         name: 'type',
         message: 'Property type:',
         type: 'list',
+        default: this.propDefinition && 
+          (Array.isArray(this.propDefinition.type) ? 
+            'array' : this.propDefinition.type),
```

Comment:
I am afraid this doesn't seem to work correctly, when I selected array of numbers in the first run, I am offered number type as the default in the second run (instead of array type).

```
$ yo loopback:property
? Select the model: Avatar
? Enter the property name: nums
? Property type: number
? Required? Yes
? Default value[leave blank for none]: asd,asd
Warning: please enter the nums property again. The default value provided "asd,asd" is not valid for the selected type: number
? Enter the property name: nums
? Property type: (Use arrow keys)
  string
❯ number
  boolean
  object
  array
  date
  buffer
(Move up and down to reveal more choices)
```